js/initcontext.go

```diff
@@ -178,6 +181,29 @@ func (m *moduleVUImpl) Runtime() *goja.Runtime {
 	return m.runtime
 }
 
+func (m *moduleVUImpl) Reserve() func(func() error) {
+	return m.eventLoop.reserve()
+}
+
+/* This is here to illustrate how to use Reserve to get a promise to work with the event loop
+// TODO move this to a common function or remove before merging
```

Comment:
Hm I guess I can move it :shrug:

Reply:
I am more for removing this - this currently here mostly for illustrative purposes - if we decided that we want it in the first version it will go in `common` more than anywhere else :thinking: . Also moving it makes it harder to explain IMO, so I think I will keep it here for now